app/src/main/java/com/framgia/soundclound/service/MusicService.java

```diff
@@ -0,0 +1,324 @@
+
+    @Override
+    public boolean onError(MediaPlayer mp, int what, int extra) {
+        return false;
+    }
+
+    @Override
+    public void onPrepared(MediaPlayer mp) {
+        mp.seekTo(0);
+        mp.start();
+        showNotifi();
+
+    }
+
+    public int getPosition() {
+        return player.getCurrentPosition();
+    }
+
+    public boolean isShuffle() {
+        return mShuffle;
+    }
+
+    public boolean isRepeat() {
+        return mRepeat;
+    }
+
+    @Override
+    public boolean isPlay() {
+        return player.isPlaying();
+    }
+
+    @Override
+    public void play() {
+        player.reset();
+        curentTrack = SharePreferences.getInstance().getIndex();
+        mTracks = SharePreferences.getInstance().getListTrack();
+        Track track = mTracks.get(curentTrack);
+        mArtist = track.getPublisherMetadata().getArtist();
+        mTitleTrack = track.getTitle();
+        mImageTrack = track.getArtworkUrl();
+        String uri = track.getUri();
+        try {
+            player.setDataSource(uri);
+            player.prepareAsync();
+        } catch (Exception e) {
+            Logger.getLogger(e.toString());
+        }
+    }
+
+
+    public boolean checkTrackPlay(Track track) {
+        Track trackCurent = mTracks.get(curentTrack);
+        return track.getUri().equals(trackCurent.getUri());
+    }
+
+    @Override
+    public void start() {
+        player.start();
+    }
+
+    @Override
+    public void pause() {
+        if (player.isPlaying()) {
+            player.pause();
+        }
+    }
+
+    @Override
+    public void stop() {
+        if (player != null) {
+            player.stop();
+        }
+    }
+
+    @Override
+    public void release() {
+        if (player != null) {
+            player.release();
+        }
+    }
+
+    @Override
+    public void seekto(int position) {
+        if (player != null) {
+            player.seekTo(position);
+        }
+    }
+
+    @Override
+    public void next() {
+        curentTrack = SharePreferences.getInstance().getIndex();
+        curentTrack++;
+        if (curentTrack > (mTracks.size() - 1)) {
+            curentTrack = 0;
+            SharePreferences.getInstance().putIndex(curentTrack);
+            play();
+        } else {
+            SharePreferences.getInstance().putIndex(curentTrack);
+            play();
+        }
+
+    }
+
+    @Override
+    public void prev() {
+        curentTrack = SharePreferences.getInstance().getIndex();
+        curentTrack--;
+        if (curentTrack < 0) {
+            curentTrack = mTracks.size() - 1;
+            SharePreferences.getInstance().putIndex(curentTrack);
+            play();
+        } else {
+            SharePreferences.getInstance().putIndex(curentTrack);
+            play();
+        }
+        SharePreferences.getInstance().putIndex(curentTrack);
+    }
+
+    @Override
+    public void onDestroy() {
+        super.onDestroy();
+        stopForeground(true);
+        stopSelf();
```

Comment:
WARNING: [CallSuperLast] super should be called at the end of the method
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/android.html#CallSuperLast